doc/source/building/build_kis.rst

```diff
@@ -0,0 +1,81 @@
+.. _kis:
+
+Build KIS Image (Kernel, Initrd, System)
+========================================
+
+.. sidebar:: Abstract
+
+   This page explains how to build an image that consists
+   out of three components. The kernel an initrd and an
+   optional root filesystem image. In {kiwi} terminology
+   this is called KIS.
+
+A KIS image is a collection of image components that are not
+associated with a dedicated use case. This means from a {kiwi}
+perspective we don't know in which environment these components
+are later used. The predecessor of this image type was called
+`pxe` under the assumption that the components will be used
+in a pxe boot environment. However this assumption is not
+neccessarily true and the image components are used in a different
+way. Because there are so many possible deployment strategies
+for a `kernel` plus `initrd` and optional `system root filesystem`,
+{kiwi} provides this as generic `KIS` type that is generically
+usable.
+
+The former `pxe` image type will continue to exist but is expected
+to be used only in combination with the legacy `netboot` infrastructure
+as described in :ref:`build_legacy_pxe`.
+
+To add a KIS build to your appliance, create a `type` element with
+`image` set to `kis` in your :file:`config.xml` as shown below:
+
+.. code:: xml
+
+   <preferences>
+       <type image="kis"/>
+   </preferences>
+
+With this image type setup {kiwi} will just build a kernel and initrd
+not associated to any system root file system. Usually such
+an image is only useful with some custom dracut extensions
+as part of the image description.
+
+The following attributes of the `type` element are often used when
+building KIS images:
+
+- `filesystem`: Specifies the root filesystem and triggers the build
+  of an additional filesystem image of that filesystem. The generated
+  kernel command line options file (append file) will then also
+  include a `root=` parameter that references this filesystem image UUID.
+  If the information from the append file should be used or not is
+  optional.
+
+- `kernelcmdline`: Specifies kernel command line options that will be
+  part of the generated kernel command line options file (append file).
+  By default the append file contains no information or the reference
+  to the root UUID if the `filesystem` attribute is used.
+
+All other attributes of the `type` element that applies to an optional
+root filesystem image will be effective in the system image of a KIS
+image as well.
+
+With the appropriate settings present in :file:`config.xml` {kiwi} can now
+build the image:
+
+.. code:: bash
+
+   $ sudo kiwi-ng --type kis system build \
+       --description kiwi-descriptions/suse/x86_64/{exc_description} \
+       --target-dir /tmp/myimage
+
+The resulting image components are saved in the folder :file:`/tmp/myimage`.
+
+.. note::
+
+   For testing the components of a KIS image the customer needs to
```

Comment:
Wouldn't make sense to include here an example of a qemu call to boot it? The only reason I see to not include it here is that the filesystem image might not be there, but isn't it still a good example for people to let them try the new type?

Reply:
Hmm, I did not include that part because we have intentionally no scope for the kis type.
Instead I added a note at the end that says:

```
.. note::

   For testing the components of a KIS image the customer needs to
   provide a deployment infrastructure and also the deployment
   process by themselve. One example of a deployment infrastructure
   using PXE is provided by {kiwi} with the `netboot` infrastructure.
   However that netboot infrastructure is no longer developed and
   only kept for compatibiliy reasons. For details
   see :ref:`build_legacy_pxe`
```

So what generic qemu testing method would you suggest ?

Reply:
I'd suggest something like

```
qemu-kvm -kernel <kernel file> -initrd <initrd file> -append $(cat <append file>) -hda <filesystem file> -m 1024
```

The only thing that is probably clearly not generic enough here is the filesystem image as this is optional. I like the note and makes sense, but probably would also make sense to provide a simple test example as the above. The above should be valid for any of our testing images (assuming a kis type is added with filesystem). I like it because is simple and makes clear the purpose of the build binaries.

I just bring this up because this is what I would have done. I am not having a strong opinion regarding that. I understand that this does not cover the use case without a filesystem. That case falls in the note you wrote.

Reply:
ok that worked for me. I have tested booting a kis image through qemu and that worked. Interesting approach to attach a rootfs file as disk but qemu managed to handle it. ok I also updated the docs regarding this part